### ISSUE TITLE
Block sites stylesheet loading as a feature for theme creators

### ISSUE THREAD

**Issue by ncla** [ncla/LoungeDestroyer#238]
So they don't have to overwrite sites style rules but can easily create their own without !important'ifying properties.

**Comment by ncla** [comment]
Related to #191

**Comment by ncla** [comment]
Added as a hidden setting at the moment.

**Comment by ncla** [comment]
mrw you can't block on disk cached requests
![](http://www.reactiongifs.com/r/2014/01/O0adam.gif)

**Comment by ncla** [comment]
Alternative more complex shit solutions:
1) Use `document.styleSheets` to check if style rules have been applied from Lounge, by checking if there is even an entry to it, and then checking the length of CSS rules. If rules exist, we could do a hard refresh with chrome.tabs.reload with bypassCache option.

2) Wipe out browser cache, ALL BROWSER CACHE, with chrome.browserData API, it does not allow to filter by domain sadly. This might be costly on performance and may produce unexpected outcomes to user.

The rest would be taken by chrome.webRequest.onBeforeRequest stylesheet blocking, it's just that we initially need to get rid of the cache for the .css files.

The 1) seems like the best viable option, now we have to consider if we want to warn users that we force refresh. Don't think we would be affecting their browsing experience by a lot since when they change the theme, they will be navigating to Lounge to check it out anyway.

**Comment by ncla** [comment]
Sadly I stumbled upon another road block to make this a successful feature. On DOTA2Lounge the stylesheet is loaded over cdn.dota2lounge.com, to which we sadly do not have permissions, we only have `*://dota2lounge.com`, not `://*.dota2lounge.com` which in hindsight is stupid.

What I did instead is if stylesheet is from cdn.*, I remove the stylesheet node manually. Not pretty and relies on DOM to be ready. :(
